src/PiWeb.Import.Sdk/Modules/ImportFormat/IImportFormat.cs

```diff
@@ -1,8 +1,51 @@
 ﻿namespace Zeiss.PiWeb.Import.Sdk.Modules.ImportFormat;
 
+#region usings
+
+using System.Collections.Generic;
+
+#endregion
+
 /// <summary>
-/// Represents a custom import format provided as part of a plugin. Custom import formats provide a way to identify the relevant files,
-/// parse the data in those files and upload the data to PiWeb Server.
+/// Represents a custom import format provided as part of a plugin. Custom import formats provide a way to identify and group relevant
+/// files, parse their data and create a data image that can be uploaded to PiWeb Server.
 /// </summary>
 public interface IImportFormat
-{ }
\ No newline at end of file
+{
+    #region methods
+
+    /// <summary>
+    /// Builds an import group if the <see cref="IImportFormat"/> recognizes the <paramref name="primaryFileSource"/> and other
+    /// <see cref="IFileSource"/>s provided by the <paramref name="context"/>.
+    /// </summary>
+    /// <param name="primaryFileSource">The file to analyze and built a group of.</param>
+    /// <param name="context">Provides context information such as other files in the import folder.</param>
+    /// <returns>An <see cref="ImportGroup"/> defining the files identified; or <c>null</c> otherwise.</returns>
+    ImportGroup? BuildGroup(IFileSource primaryFileSource, IGroupContext context);
+
+    #endregion
+
+    #region properties
+
+    /// <summary>
+    /// Gets the format identifier.
```

Comment:
I think when writing code comments, we should treat properties like members, not like methods. So we would just describe what it is without using a verb. In this case:
"The format identifier."